### PR TITLE
docs: transition v4 to v5

### PR DESCRIPTION
**Pull request by PhilippeR26**
## Motivation and Resolution

Solve #539  and #564 
Add in the guides a page to explain where users have to change their code using Starknet.js v4.

**Could you comment if you identify other subjects?**

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [ ] Documented the changes in code (API docs will be generated automatically)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

File: www/docs/guides/transition.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+```typescript
+// v4
+const qty = new BN("0x4a8bc");
+
+// v5
+const qty1 = BigInt("0x4a8bc");
+const qty2 = 32786324915918425n;
+```
+
+So, the BigNumberish type is now defined as :
```
Comment (ivpavici) [inline www/docs/guides/transition.md:38]:
```suggestion
So, the `BigNumberish` type is now defined as :
```

File: www/docs/guides/transition.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+- String representing a number : "123", "0xabc2"
+- Number (max 53 bits) : 123
+- BigInt (max 255 bits): 12345612345n
+
+All methods of BN type are of course removed, but have equivalences :
+
+- .mul() and .div() : common operators can be used with bigints : + - \* /
+- .umod() : can easily be replaced :
```
Comment (ivpavici) [inline www/docs/guides/transition.md:47]:
```suggestion
- .umod() : can easily be replaced with :
```

File: www/docs/guides/transition.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+```
+
+- .toarray() : can be replaced by `num.hexToBytes(a)`.
+
+## ec (elliptic curve)
+
+BN elliptic curves library has been replaced by a bigint library.  
+Many functions have been relocated/modified.
+The concept of key pair has been removed :  
+`ec.getKeyPair(privateKey)`, `ec.getKeyPairFromPublicKey(pubKey)` and `ec.genKeyPair()` do not exists anymore. Methods are using now directly the private key :
```
Comment (ivpavici) [inline www/docs/guides/transition.md:65]:
```suggestion
`ec.getKeyPair(privateKey)`, `ec.getKeyPairFromPublicKey(pubKey)` and `ec.genKeyPair()` do not exist anymore.
Methods are now using the private key directly:
```

File: www/docs/guides/transition.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+It has been considered advantageous to replace `DeclareDeploy` by `DeclareAndDeploy` :
+
+```typescript
+// v4
+const response = await account0.declareDeploy({ contract: compiledTest, classHash: testClassHash });
+// v5
+const response = await account0.declareAndDeploy({ contract: compiledHelloSierra, casm: compiledHelloCasm });
+```
+
+> Nota : `declare` and `declareAndDeploy` do not needs anymore the classHash ; the new ec library is now able to calculate it quickly and automatically. If the contrat has been written in Cairo 1, the casm property is needed.
```
Comment (ivpavici) [inline www/docs/guides/transition.md:134]:
```suggestion
> Note : `declare` and `declareAndDeploy` do not need anymore the classHash! The new ec library is now able to calculate it quickly and automatically. If the contract has been written in Cairo 1, the casm property is needed.
```

**Comment by ivpavici** [comment]
@penovicp if you don't have any more suggestions we can merge this!

**Review by penovicp — APPROVED** [review]

**Pull request merged**